### PR TITLE
CMPS-167

### PR DESCRIPTION
**Pull request by YuriiFetko**

### CODE REVIEW

File: Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts [hunk 1]
```diff
@@ -19,4 +36,108 @@ export class CalendarComponent {
...
+
+  ngAfterViewInit() {
+    this.calendarComponentApi = this.calendarComponent.getApi();
+  }
+
+  onOpenDialog(): void {
+    const dialogRef = this.dialog.open(EventModalComponent, {
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts:62]: Use `OnDestroy` event to unsubscribe from Observable

File: Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts [hunk 2]
```diff
@@ -19,4 +36,108 @@ export class CalendarComponent {
...
+  }
+
+  dateClick(calendarEvent) {
+    const dialogRef = this.dialog.open(EventModalComponent, {
+      data: {
+        date: calendarEvent.dateStr
+      }
+    });
+
+    dialogRef.afterClosed().subscribe(result => {
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts:76]: Use `OnDestroy` event to unsubscribe from Observable

File: Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts [hunk 3]
```diff
@@ -19,4 +36,108 @@ export class CalendarComponent {
...
+    }
+
+  }
+
+  editEvent(info) {
+    const dialogEditRef = this.dialog.open(EditEventComponent, {
+      data: info.event
+    });
+
+    dialogEditRef.afterClosed().subscribe(result => {
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts:127]: Use `OnDestroy` event to unsubscribe from Observable

File: Src/Campus.Master.API/UI/src/app/modules/calendar/event-modal/create-event/event-modal.component.html [hunk 4]
```diff
@@ -0,0 +1,133 @@
+<form class="dialog" [formGroup]="dialogForm" (ngSubmit)="createEvent()">
+  <div class="dialog__header">
+    <div class="dialog__header_wrapper">
+      <div class="dialog__ico"></div>
+      <div>
+        <h1 class="dialog__title">Створити подію</h1>
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/modules/calendar/event-modal/create-event/event-modal.component.html:6]: Translate page with i18n

File: Src/Campus.Master.API/UI/src/app/modules/calendar/event-modal/delete-event/delete-event.component.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
+<form class="dialog">
+  <div class="dialog__header">
+    <div class="dialog__header_wrapper">
+      <div class="dialog__ico"></div>
+      <div>
+        <h1 class="dialog__title">Видалити подію</h1>
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/modules/calendar/event-modal/delete-event/delete-event.component.html:6]: Translate page with i18n

File: Src/Campus.Master.API/UI/src/app/modules/calendar/event-modal/edit-event/edit-event.component.html [hunk 6]
```diff
@@ -0,0 +1,123 @@
+<form class="dialog" [formGroup]="dialogForm" (ngSubmit)="createEvent()">
+  <div class="dialog__header">
+    <div class="dialog__header_wrapper">
+      <div class="dialog__ico"></div>
+      <div>
+        <h1 class="dialog__title">Редагувати подію</h1>
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/modules/calendar/event-modal/edit-event/edit-event.component.html:6]: Translate page with i18n

File: Src/Campus.Services/Core/EventService.cs [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+            await _context.Entry(user)
+                .Collection(u => u.Participation)
+                .LoadAsync();
+
+            Participant defaultParticipation = user.Participation.FirstOrDefault();
+
+            if (defaultParticipation == null)
+                throw new ApplicationException("Given user doesn't participate in any classroom");
+
+            await _context.Entry(defaultParticipation)
```
Comment (antonAce) [inline Src/Campus.Services/Core/EventService.cs:136]: `Recommendation`: Merge async loads with [Task.WhenAll()](https://jeremylindsayni.wordpress.com/2019/03/11/using-async-await-and-task-whenall-to-improve-the-overall-speed-of-your-c-code/)

File: Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.html [hunk 8]
```diff
@@ -1,11 +1,11 @@
 <div class="calendar__container">
   <mat-card>
     <div class="calendar__header">
-      <h1>Calendar</h1>
-      <button class="mat-raised-button">+ New event</button>
+      <h1>Календар</h1>
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.html:4]: Translate page with i18n

File: Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts [hunk 9]
```diff
@@ -1,14 +1,31 @@
...
+  @ViewChild('calendar') calendarComponent: FullCalendarComponent;
+  public isUpdatedEvent = false;
+
+  private calendarComponentApi;
   calendarOptions: CalendarOptions = {
+    dateClick: this.dateClick.bind(this),
+    eventClick: this.editEvent.bind(this),
+    eventChange: this.editEventFromCalendar.bind(this),
+    events: this.events,
+    locale: 'uk',
```
Comment (antonAce) [inline Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts:27]: You can assign automatically detected locale from `LocaleService`

File: Src/Campus.Services/Core/EventService.cs [hunk 10]
```diff
@@ -0,0 +1,147 @@
...
+
+                eventViews.Add(eventViewDto);
+            }
+
+            return eventViews;
+        }
+
+        public async Task<int> AddEvent(string userId, EventAddDto eventDto)
+        {
+            Classroom defaultClassroom = await GetDefaultClassroomByUserId(userId);
```
Comment (antonAce) [inline Src/Campus.Services/Core/EventService.cs:66]: Use `var` instead

File: Src/Campus.Services/Core/EventService.cs [hunk 11]
```diff
@@ -0,0 +1,147 @@
...
+            defaultClassroom.Events.Add(newEvent);
+
+            await _context.SaveChangesAsync();
+
+            return newEvent.Id;
+        }
+
+        public async Task EditEventById(string userId, EventEditDto eventDto)
+        {
+            Classroom defaultClassroom = await GetDefaultClassroomByUserId(userId);
```
Comment (antonAce) [inline Src/Campus.Services/Core/EventService.cs:87]: Use `var` instead

File: Src/Campus.Services/Core/EventService.cs [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+
+            Event eventToEdit = defaultClassroom.Events
```
Comment (antonAce) [inline Src/Campus.Services/Core/EventService.cs:89]: Use `var` instead

File: Src/Campus.Services/Core/EventService.cs [hunk 13]
```diff
@@ -0,0 +1,147 @@
...
+            eventToEdit.AllDay = eventDto.AllDay;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task DeleteEventById(string userId, string eventId)
+        {
+            Classroom defaultClassroom = await GetDefaultClassroomByUserId(userId);
+
+            Event eventToDelete = defaultClassroom.Events
```
Comment (antonAce) [inline Src/Campus.Services/Core/EventService.cs:109]: Use `var` instead

File: Src/Campus.Services/Core/EventService.cs [hunk 14]
```diff
@@ -0,0 +1,147 @@
...
+            eventToEdit.EndDate = eventDto.End;
+            eventToEdit.ActualLocation = eventDto.Location;
+            eventToEdit.AllDay = eventDto.AllDay;
+
+            await _context.SaveChangesAsync();
+        }
+
+        public async Task DeleteEventById(string userId, string eventId)
+        {
+            Classroom defaultClassroom = await GetDefaultClassroomByUserId(userId);
```
Comment (antonAce) [inline Src/Campus.Services/Core/EventService.cs:107]: Use `var` instead

File: Src/Campus.Services/Core/EventService.cs [hunk 15]
```diff
@@ -0,0 +1,147 @@
...
+
+        public EventService(UserManager<User> userManager, CampusContext context)
+        {
+            _userManager = userManager;
+            _context = context;
+        }
+
+        public async Task<IEnumerable<EventViewDto>> GetClassroomEventsByUserId(string userId)
+        {
+            Classroom classroom = await GetDefaultClassroomByUserId(userId);
```
Comment (antonAce) [inline Src/Campus.Services/Core/EventService.cs:27]: Use `var` instead

**Review by antonAce — COMMENTED** [review]
Some technical issues and recommendations

File: Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts (hunk 1 above)
Comment (YuriiFetko) [reply Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts]: fixed

File: Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts (hunk 2 above)
Comment (YuriiFetko) [reply Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts]: fixed

File: Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts (hunk 3 above)
Comment (YuriiFetko) [reply Src/Campus.Master.API/UI/src/app/modules/calendar/calendar.component.ts]: fixed

File: Src/Campus.Master.API/UI/src/app/modules/calendar/event-modal/delete-event/delete-event.component.html (hunk 5 above)
Comment (antonAce) [reply Src/Campus.Master.API/UI/src/app/modules/calendar/event-modal/delete-event/delete-event.component.html:6]: `Moved to other task`

**Review by profileEnumerable — APPROVED** [review]
📦

**Pull request merged**